core/src/test/scala/sbt/librarymanagement/VersionRangeSpec.scala

```diff
@@ -10,7 +10,7 @@ class VersionRangeSpec extends UnitSpec {
   it should "strip [1.0,) to 1.0" in stripTo("[1.0,)", Some("1.0"))
   it should "strip (1.0,) to 1.0" in stripTo("(1.0,)", Some("1.0"))
   it should "strip (1.0,2.0) to None" in stripTo("(1.0,2.0)", None)
-  it should "strip [1.0,2.0] to None" in stripTo("[1.0,2.0]", None)
+  it should "strip [1.0,2.0] to 2.0" in stripTo("[1.0,2.0]", Some("2.0"))
```

Comment:
Upper bound could be a non-existent version, so I think it's usually better to prioritize to the lower bound to keep things within the graph.